modules/python.nix

```diff
@@ -0,0 +1,145 @@
+{ pkgs, ... }:
+let
+  pip = pkgs.callPackage ../pkgs/pip { };
+
+  poetry = pkgs.callPackage ../pkgs/poetry { };
+
+  python = pkgs.python310Full;
+
+  stderred = pkgs.replitPackages.stderred;
+
+  prybar = pkgs.replitPackages.prybar-python310;
+
+  pypkgs = pkgs.python310Packages;
+
+  debugpy = pypkgs.debugpy;
+
+  dapPython = pkgs.replitPackages.dapPython;
+
+  python-lsp-server = pkgs.callPackage ../pkgs/python-lsp-server { };
+
+  python-ld-library-path = pkgs.lib.makeLibraryPath [
+    # Needed for pandas / numpy
+    pkgs.stdenv.cc.cc.lib
+    pkgs.zlib
+    # Needed for pygame
+    pkgs.glib
+    # Needed for matplotlib
+    pkgs.xorg.libX11
+  ];
+
+  python3-wrapper = pkgs.stdenvNoCC.mkDerivation {
+    name = "python3-wrapper";
+    buildInputs = [ pkgs.makeWrapper ];
+
+    buildCommand = ''
+      mkdir -p $out/bin
+      makeWrapper ${python}/bin/python3 $out/bin/python3 \
+        --set LD_LIBRARY_PATH "${python-ld-library-path}"
```

Comment:
what does this do? does it set the `LD_LIBRARY_PATH` env variable every time one runs `python3`?

some things broke when we did that outside of nix, but given that this is being done _inside_ it might be different. how are we testing this?

Reply:
It creates this shell script:
```
#! /nix/store/dsd5gz46hdbdk2rfdimqddhq6m8m8fqs-bash-5.1-p16/bin/bash -e
export LD_LIBRARY_PATH='/nix/store/mdck89nsfisflwjv6xv8ydj7dj0sj2pn-gcc-11.3.0-lib/lib:/nix/store/026hln0aq1hyshaxsdvhg0kmcm6yf45r-zlib-1.2.13/lib:/nix/store/2k366jrbsra97gjfxwvrhvixjfxdach5-glib-2.74.1/lib:/nix/store/w3zzhfl4a7xp0xfflz2gawv02y8ba9z8-libX11-1.8.1/lib'
exec "/nix/store/hd4cc9rh83j291r5539hkf6qd8lgiikb-python3-3.10.8/bin/python3"  "$@"
```

Reply:
Yeah, I know the bug. This is isolated from the user's environment, so it should be fine.

Reply:
i suspect that it still may cause issues: if there's a DSO bundled into a package that is linked against another version of glibc we may see linker errors. if we test enough edge cases where there are `.so` files in the wheel, we're good.